l1-contracts/src/core/RollupCore.sol

```diff
@@ -35,29 +35,171 @@ import {StakingQueueConfig} from "@aztec/core/libraries/compressed-data/StakingQ
+ *      - `setupEpoch`: Called to initialize the validator selection for the next epoch
+ *
+ *      The block building flow is as follows:
+ *
+ *      - At each L2 slot a single proposer is chosen from the validator set who assembles a block that:
+ *         - Builds on top of the last pending block (tips.pending) in the rollup
+ *         - Includes state transitions, messages, and fee calculations
+ *         - Contains proper archive linking to maintain chain continuity
+ *         - Is attested by at least 2/3 of the committee members
+ *      - The L2 block is posted to L1 via the `propose` function
+ *         - The pending chain tip advances to the new block
+ *         - State is updated (message trees, archives, etc.)
+ *      - After the epoch has ended, a prover generates a proof of the state transition for all blocks in the epoch
+ *         - The proof is submitted via `submitEpochRootProof`
+ *         - The proof must be submitted within the configured proof submission window
+ *         - Upon successful proof submission the proven chain tip advances to the last block in the proven epoch
+ *      - Proving a block makes it finalized from the perspective of L1
+ *         - This triggers disbursing rewards to the sequencers and provers of the epoch
+ *         - And pushes messages to the outbox for L1 processing
+ *
+ *      Unhappy path for invalid attestations:
+ *      - Attestations in blocks are not validated on-chain to save gas. Since attestations are still posted to L1,
+ *        nodes are expected to verify them off-chain, and skip a block if its attestations are invalid.
+ *      - If a block has invalid attestation signatures, anyone can call `invalidateBadAttestation()`
+ *      - If a block has insufficient valid attestations (< 2/3 of committee), anyone can call
+ *        `invalidateInsufficientAttestations()`
+ *      - While anyone can call invalidation functions, it is expected that the next proposer will do so, and if they
+ *        fail to do so, then other committee members do, and if they fail to do so, then any validator will do so.
+ *      - Upon invalidation, the invalid block and all subsequent blocks are removed from the chain, so the pending
+ *        chain tip reverts to the block immediately before the invalid one.
+ *       - Note that only unproven blocks can be invalidated, as proven blocks are final and cannot be reverted.
+ *
+ *      Unhappy path for missing proofs:
+ *      - Each epoch has a proof submission window (configured via aztecProofSubmissionEpochs).
+ *      - If no proof is submitted within the window, it is assumed that the epoch cannot be proven due to missing data,
+ *        so all blocks in the epoch are pruned. This is done by calling `prune()` manually, or automatically on the
+ *        next proposal.
+ *      - The committee for the epoch is expected to disseminate transaction data to allow proving, so a prune is
+ *        considered a slashable offense,
```

Comment:
This is interesting. What is the best place to learn about what this committee data dissemination entails?

Reply:
Its sharing the transactions and proofs on p2p.